src/collection.js

```diff
@@ -74,13 +74,10 @@ export class SyncResultObject {
     }
     // Deduplicate entries by id. If the values don't have `id` attribute, just
     // keep all.
-    const deduplicated = this[type].concat(entries).reduce((acc, cur) => {
-      const existing = acc.filter(
-        r => (cur.id && r.id ? cur.id != r.id : true)
-      );
-      return existing.concat(cur);
-    }, []);
-    this[type] = deduplicated;
+    const ids = new Set(entries.map(({ id }) => id));
+    this[type] = this[type]
+      .filter(({ id }) => !id || !ids.has(id))
+      .concat(entries);
```

Comment:
What if we see `result.add("skipped", [{id: 1, name: "a"}, {id: 2, name: "b"}])`?